lib/luahid.c

```diff
@@ -153,6 +173,64 @@ static int luahid_probe(struct hid_device *hdev, const struct hid_device_id *id)
 	return hid_hw_start(hdev, HID_CONNECT_DEFAULT);
 }
 
+static int luahid_doreport_fixup(lua_State *L, luahid_t *hid,
+				 struct hid_device *hdev, __u8 *buf, unsigned int *size,
+				 __u8 **ret_ptr)
+{
+	if (luahid_checkdriver(L, hid, -1, "_report_des")) {
+		pr_err("report_fixup: couldn't find driver\n");
+		return -ENXIO;
+	}
+
+	if (lua_getfield(L, -2, "report_fixup") != LUA_TFUNCTION)
+		return -ENXIO;
+
+	lua_pushvalue(L, -3); /* hid.ops */
+	luahid_push_device_info(L, hdev);
+	lunatik_object_t *original_data = luadata_new(buf, *size, hid->runtime->sleep, LUADATA_OPT_READONLY);
```

Comment:
if we don't have a practical case on increasing the size of the buffer, let's just let the user change the original data in place.. instead of creating and returning a new data object.. (even if we have a practical case, but it's not common, I would implement the simplified version firstly and create a issue for handling this case later)

Reply:
we should add support on luadata for this sort of dupping.. it's kinda straightforward to implement.. something as, `data:dup([size])`

Reply:
and I don't think it's necessary to the original to be READONLY, right? if it's not necessary to change its size, we can just return the same object, right/